core/domain/stats_jobs_one_off.py

```diff
@@ -18,14 +18,403 @@
 import collections
 
 from core import jobs
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import stats_domain
 from core.domain import stats_jobs_continuous
+from core.domain import stats_services
 from core.platform import models
 
-(stats_models,) = models.Registry.import_models([
-    models.NAMES.statistics
+import feconf
+import utils
+
+(stats_models, exp_models) = models.Registry.import_models([
+    models.NAMES.statistics, models.NAMES.exploration
 ])
 
 
+class MigrateStatistics(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off migration of existing event model instance to generate
+    ExplorationStatsModel instances for all existing explorations.
+    """
+
+    EVENT_TYPE_STATE_ANSWERS = 'state_answers'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            stats_models.StartExplorationEventLogEntryModel,
+            stats_models.CompleteExplorationEventLogEntryModel,
+            stats_models.StateHitEventLogEntryModel,
+            stats_models.StateAnswersModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function. Must be declared @staticmethod.
+
+        Args:
+            item: StartExplorationEventLogEntryModel,
+                CompleteExplorationEventLogEntryModel,
+                StateHitEventLogEntryModel or StateAnswersModel.
+
+        Yields:
+            tuple. For StartExplorationEventLogEntryModel and
+                CompleteExplorationEventLogEntryModel, a 2-tuple of the form
+                (exp_id, value).
+                The structure of 'value' is:
+                    {
+                        'event_type': str. Type of event.
+                        'session_id': str. ID of the session.
+                    }
+            tuple. For StateHitEventLogEntryModel, a 2-tuple of the form
+                (exp_id, value).
+                The structure of 'value' is:
+                    {
+                        'event_type': str. Type of event.
+                        'version': str. Version of the exploration.
+                        'state_name': str. Name of the state.
+                        'session_id': str. ID of the session.
+                        'created_on': How many milliseconds ago the event was
```

Comment:
Shouldn't this be measured using "milliseconds since the Epoch" (or some other fixed point in time, but the Epoch is the default) rather than "how many milliseconds ago"? The latter number changes continuously over time.